docs/user-guide/configuring.md

```diff
@@ -713,6 +713,61 @@ module.exports = {
 }
 ```
 
+## Configuration Based on Glob Patterns
+
+Sometimes a more fine-controlled configuration is necessary, for example if the configuration for files within the same directory has to be different. Therefore you can provide configurations under the `overrides` key that will only apply to files that match specific glob patterns, using the same format you would pass on the command line (e.g., `app/**/*.test.js`).
+
+### How it works
+
+* Glob pattern overrides can only be configured within `.eslintrc` files.
```

Comment:
This line is a bit confusing, since overrides can be configured in any config files (such as `.eslintrc.json`, `.eslintrc.yml`, or the `eslintConfig` property of a `package.json` file), not just `.eslintrc` files.

On a related note, since `.eslintrc` is deprecated in favor of the versions with file extensions, maybe we should use file extensions in the examples.